### DISCUSSION TITLE
Search

### DISCUSSION THREAD

**Discussion by gummipunkt** [neet/masto.js#846] [Q&A]
Hi,

with the following search string, I only receive results from my instance:
```js
const search = await masto.v2.search({
        q: 'hashtags[' + hashtag + ']'});

console.log(search);
```

with the URL https://instance.tld/api/v2/search?q=hashtags[hashtag]

I'll find search results from all known instances. What do I wrong here?

**Comment by neet** [comment]
Masto.js uses exactly the same API as Mastodon web UI and does not undertake anything special for you. If you search by the same query on the web UI and result in the same statuses, it is highly likely that something is wrong in your query string.

It would be helpful to take a look at Mastodon's official documentation. It says:

> Mastodon’s full-text search allows logged in users to find results from their own statuses, their mentions, their favourites, and their bookmarks. It deliberately does not allow searching for arbitrary strings in the entire database.
> https://docs.joinmastodon.org/admin/optional/elasticsearch/

In other words, the result will only show you the statuses that you have interacted with. It doesn't act like looking up the entire timeline retroactively.

I hope it helps

**Reply by neet** [reply]
P.S. The code you provided looks like you're trying to get statuses that include a specific hashtag. In that case, you may want to use `v1.timelines.listHashtag` like in below

```js
const masto = login(...);
const statuses = await masto.timelines.listHashtag("mastojs");
```

**Reply by gummipunkt** [reply]
Thank you, unfortunately, it doesn't work:
`TypeError: Cannot read properties of undefined (reading 'listHashtag')`

**Reply by gummipunkt** [reply]
const statuses = await masto.v1.timelines.listHashtag("mastojs");

With v1 it works ;-)

Thank you

**Reply by neet** [reply]
> With v1 it works ;-)

Ah yes, this is what I meant. I'm glad to hear it's resolved now 👍
